pamelalozano/src/main/java/com/google/sps/servlets/Auth.java

```diff
@@ -0,0 +1,63 @@
+
+package com.google.sps.servlets;
+
+
+import java.util.*;  
+import com.google.gson.Gson;
+import java.io.IOException;
+
+public class Auth {
```

Comment:
FYI mostly, as this is totally fine as is:

Best practice here would be to make all of these final fields. The way to do this would be to create a Auth.Builder that can add all of the fields individually and then call ".build()" to actually create the Auth instance. At Google, we would use a code generation annotation to do this called AutoValue. Here are the docs on how to use it! https://github.com/google/auto/blob/master/value/userguide/index.md

Reply:
So you mean something like this right?
https://www.vogella.com/tutorials/DesignPatternBuilder/article.html

Reply:
Hand written builder example. This is very boring & error-prone to write so we can make AutoValue generate this. 

```java
public class Auth {
  private final boolean isLoggedIn;
  // More private final fields

  // Public getter for isLoggedIn  
  public boolean getIsLoggedIn() {
    return isLoggedIn;
  }
  // Other getters. No setters!

  // Auth constructor is private.
  // There is no way to create an instance of Auth without using the Builder
  private Auth(Builder builder) {
    isLoggedIn = builder.isLoggedIn;
    // Set other Auth final fields from builder 
  }

  // static inner class can be instantiated like `new Auth.Builder()`
  public static class Builder {
    private isLoggedIn;
    // Other private (not final) fields internal to Builder
    
    public Builder setIsLoggedIn(boolean isLoggedIn) {
      this.isLoggedIn = isLoggedIn;
      return this;
    }

    // Other "set..." methods to set internal Builder fields

    public Auth build() {
      return new Auth(this);
    }
  }
}
```

Usage
```java
Auth myAuth = new Auth.Builder().setIsLoggedIn(true).build();
System.out.println("User logged in?: " + myAuth.getIsLoggedIn());
```

Then the obvious issue is "What if I have an instance of `Auth` but I need to update one value?". You write a method on Auth `Builder toBuilder() {...}` that returns a new builder with all of the fields copied and use it like this 
```java
Auth updatedAuth = oldAuth.toBuilder().setWhatever(newValue).build();
```

AutoValue can also generate `toBuilder()` for you.

Reply:
Becomes this when using AutoValue.

```java
import com.google.auto.value.AutoValue;

@AutoValue
abstract class Auth {
  abstract boolean isLoggedIn();

  // So you can get a builder like `Auth.builder()` instead of `new AutoValue_Auth.Builder()`
  static Builder builder() {
    return new AutoValue_Auth.Builder();
  }

  @AutoValue.Builder
  abstract static class Builder {
    abstract Builder setIsLoggedIn(boolean isLoggedIn);
    abstract Auth build();
  }
}
```

Reply:
Thank you so much Brett this is super helpful!

Reply:
Done!